packages/core/src/factory.ts

```diff
@@ -37,15 +39,31 @@ export function factory<Resource, Input, Output, Config, LoadOpts>(
         // build the config for this request
         const configMerger = configMergerFactory(defaultConfig, customConfig, c);
         const configObj: Config | undefined = configMerger(input);
+        let inputValidations: IPrismDiagnostic[] = [];
 
         // find the correct resource
         let resource: Resource | undefined;
         if (components.router) {
-          resource = components.router.route({ resources, input, config: configObj }, defaultComponents.router);
+          try {
```

Comment:
`process` function starts to look very scary. It's very lengthy and full of mutations. If we keep adding to it, it's gonna be unmanageable (or it is already). @chris-miaskowski, can you do a bit of scouting and refactor this before it explodes?

Reply:
This is a piece of code that we don't want to touch right now. You are right that it requires refactoring  but it's a rabbit hole - trust my word on it, I've been here were this function was created.

The best I could think of right now is wrapping some of the internals in helper functions but not sure is that any good. It's kind of like touching linux core ;)

Let me know if you have any specific refactoring suggestions, I eager to hear your thoughts on that.

Reply:
Yeah, I totally understand. But even the comments inside, they clearly show that these should be separate functions. Why even have these comments? Distinctive function names would make it clear enough what's being done. So, at least, we could change all the `let`s to `const`s and have these consts populated in this way `const x = getX(...)`. So, for example, instead of:
```typescript
// build output
        let output: Output | undefined;
        if (resource && components.mocker && (configObj as IPrismConfig).mock) {
```
let's have:
`const output = buildOutput(...)`

Reply:
Yeah, I think once we document the HTTP Client API (this sprint?) we are going to start wanting to make the API nicer too, and refactoring the guts can be part of that general cleanup.